test/symbol_name/nmcheck_prefix.pl

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/perl
+
+# Copyright (c) 2017      IBM Corporation. All rights reserved.
+
+# env var MYBASE should be the top dir where ompi/ opal/ orte/ test/ etc live.
+# env var OMPI_LIBMPI_NAME should be @OMPI_LIBMPI_NAME@ from automake
+#
+# Most likely the libs we want to check are
+#     ompi/.libs/lib<mpi>.so
+#     ompi/mpi/fortran/mpif-h/.libs/lib<mpi>_mpifh.so
+#     ompi/mpi/fortran/use-mpi-tkr/.libs/lib<mpi>_usempi.so
+#     orte/.libs/libopen-rte.so
+#     opal/.libs/libopen-pal.so
+# but I hate to assume those are the locations, so I'll use 'find' and
+# test whatever ".so"s are found.
+
+@libs = ();
+$mpi = $ENV{OMPI_LIBMPI_NAME};
+for $name (
+    "lib${mpi}.so",
+    "lib${mpi}_mpifh.so",
+    "lib${mpi}_usempi.so",
+    "libopen-rte.so",
+    "libopen-pal.so" )
+{
+    for $loc (split(/\n/, `find $ENV{MYBASE} -name $name`)) {
+        push @libs, $loc;
+    }
+}
+
+print "Checking for bad symbol names in the main libs:\n";
+$isbad = 0;
+for $lib (@libs) {
+    print "checking $lib\n";
+    check_lib_for_bad_exports($lib);
+}
+if ($isbad) { exit(-1); }
+exit(0);
+
+
+sub check_lib_for_bad_exports {
+    my $lib = $_[0];
+    my @symbols;
+    my $s;
+
+    @symbols = get_nm($lib, 'all');
+
+    # grep to get rid of symbol prefixes that are considered acceptable,
+    # leaving behind anything bad:
+    @symbols = grep(!/^ompi_/i, @symbols);
+    @symbols = grep(!/^opal_/i, @symbols);
+    @symbols = grep(!/^orte_/i, @symbols);
+    @symbols = grep(!/^orted_/i, @symbols);
+    @symbols = grep(!/^mpi_/i, @symbols);
+    @symbols = grep(!/^pmpi_/i, @symbols);
+    @symbols = grep(!/^pmix_/i, @symbols);
+    @symbols = grep(!/^mpit_/, @symbols);
+    @symbols = grep(!/^ompit_/, @symbols);
+    @symbols = grep(!/^MPIR_/, @symbols);
+    @symbols = grep(!/^MPIX_/, @symbols);
+    @symbols = grep(!/^mpidbg_dll_locations$/, @symbols);
+    @symbols = grep(!/^mpimsgq_dll_locations$/, @symbols);
+
+    @symbols = grep(!/^mca_/, @symbols); # I'm tempted to call these bad
+    @symbols = grep(!/^netpatterns_/, @symbols); # I'm tempted to call these bad
+
+    @symbols = grep(!/^_fini$/, @symbols);
+    @symbols = grep(!/^_init$/, @symbols);
+    @symbols = grep(!/^_edata$/, @symbols);
+    @symbols = grep(!/^_end$/, @symbols);
+    @symbols = grep(!/^__bss_start$/, @symbols);
+    @symbols = grep(!/^__malloc_initialize_hook$/, @symbols);
+
+    # This is another section I'm not confident about, I'm not
+    # familiar with libevent so for now I'm giving everything related
+    # to it a pass. I'd like to change all these to opal_event_* but
+    # not without confirmation from someone else first.
+    @symbols = grep(!/^_event_debug_mode_on$/, @symbols);
+    @symbols = grep(!/^event_enable_debug_output$/, @symbols);
+    @symbols = grep(!/^event_global_current_base_$/, @symbols);
+    @symbols = grep(!/^event_module_include$/, @symbols);
+    @symbols = grep(!/^eventops$/, @symbols);
```

Comment:
@rhc54 Can you confirm: I think that these should probably all be prefixed with `opal_`.  I.e., we shouldn't be exposing any public symbols from libevent that aren't prefixed.